adapters/discord/src/message.ts

```diff
@@ -1,24 +1,76 @@
-import { Dict, Messenger, Schema, segment } from '@satorijs/satori'
+import { Dict, Messenger, Schema, segment, Universal, Session, Quester } from '@satorijs/satori'
 import FormData from 'form-data'
 import { DiscordBot } from './bot'
-import { adaptMessage } from './utils'
+import { Channel, Message } from './types'
+import { adaptMessage, sanitize } from './utils'
 
 type RenderMode = 'default' | 'figure'
 
+class State {
+  author: Partial<Universal.Author> = {}
+  quote: Partial<Universal.Message> = {}
+  channel: Partial<Channel> = {}
+  fakeMessageMap: Record<string, Session[]> = {} // [userInput] = discord messages
+  threadCreated = false // forward: send the first message and create a thread
+
+  constructor(public type: 'message' | 'forward') { }
+}
+
 export class DiscordMessenger extends Messenger<DiscordBot> {
+  private stack: State[] = [new State('message')]
   private buffer: string = ''
   private addition: Dict = {}
   private figure: segment = null
   private mode: RenderMode = 'default'
 
+  get webhook() {
+    return this.bot.webhooks[this.channelId]
+  }
+  set webhook(val) {
+    if (!val) {
+      delete this.bot.webhookLock[this.channelId]
+    }
+    this.bot.webhooks[this.channelId] = val
+  }
+
   async post(data?: any, headers?: any) {
     try {
-      const result = await this.bot.http.post(`/channels/${this.channelId}/messages`, data, { headers })
+      let url = `/channels/${this.channelId}/messages`
+      if (this.stack[0].author.nickname || this.stack[0].author.avatar || (this.stack[0].type === 'forward' && !this.stack[0].threadCreated)) {
+        let webhook = await this.ensureWebhook()
+        url = `/webhooks/${webhook.id}/${webhook.token}?wait=true`
+      }
+      if (this.stack[0].type === 'forward' && this.stack[0].channel?.id) {
+        // 发送到子区
+        if (this.stack[1].author.nickname || this.stack[1].author.avatar) {
+          let webhook = await this.ensureWebhook()
+          url = `/webhooks/${webhook.id}/${webhook.token}?wait=true&thread_id=${this.stack[0].channel?.id}`
+        } else {
+          url = `/channels/${this.stack[0].channel.id}/messages`
+        }
+      }
+      const result = await this.bot.http.post<Message>(url, data, { headers })
       const session = this.bot.session()
-      await adaptMessage(this.bot, result, session)
+      const message = await adaptMessage(this.bot, result, session)
       session.app.emit(session, 'send', session)
       this.results.push(session)
+
+      if (this.stack[0].type === 'forward' && !this.stack[0].threadCreated) {
+        this.stack[0].threadCreated = true
+        let thread = await this.bot.internal.startThreadFromMessage(this.channelId, result.id, {
+          name: 'Forward',
+          auto_archive_duration: 60,
+        })
+        this.stack[0].channel = thread
+      }
+
+      return message
     } catch (e) {
+      if (Quester.isAxiosError(e) && e.response.data.code === 10015) {
```

Comment:
```suggestion
      if (Quester.isAxiosError(e) && e.response?.data.code === 10015) {
```

如果是请求阶段的错误会没有 `e.response`。